app/controllers/general.php

```diff
@@ -155,6 +156,70 @@ function router(App $utopia, Database $dbForPlatform, callable $getProjectDB, Sw
             $path .= '?' . $query;
         }
 
+        $protocol = $request->getProtocol();
+
+        // Ensure preview authorization
+        $requirePreview = \is_null($apiKey) || !$apiKey->isPreviewAuthDisabled();
+        if ($isPreview && $requirePreview) {
+            $cookie = $request->getCookie(Auth::$cookieNamePreview, '');
+            $authorized = false;
+
+            // Security checks to mark authorized true
+            if (!empty($cookie)) {
+                $jwt = new JWT(System::getEnv('_APP_OPENSSL_KEY_V1'), 'HS256', 3600, 0);
+
+                $payload = [];
+                try {
+                    $payload = $jwt->decode($cookie);
+                } catch (JWTException $error) {
+                    // Authorized remains false
```

Comment:
Shouldn't we throw here an exception?

Reply:
No need, `$payload` got default value (empty array), so it will simply end up with `$authorized=false`.

If we throw, we show error page. But instead we want logic for when $authorized is false, which is redirecting to Console to get new JWT